### DISCUSSION TITLE
Sum of Payload parameter

### DISCUSSION THREAD

**Discussion by Pernic** [TelemetryDeck/Viewer#233] [Q&A] [answered]
Hello,

I have a signal for an order, with into the payload the number of items and their type like that : 
Payload = {"enter" : 2, "main" : 3, "dessert" : 2}.

I would like to make a custom parameter to make the sum of dessert for all the order. 
Do you have any idea on how to do this ?

Thanks

**Comment by winsmith** [comment, marked as answer]
Hi, all payload keys are of type `String` so this won't work. My recommendation is to use separate signals per course and use the `floatValue` parameter, which you can sum up. 

This is what your signal could look like:

```json
{
  "type": "order.dessertSum",
  "payload": {},
  "floatValue": 2
}
```

Then your (custom) query could look like this:

```json
{
  "aggregations": [
    {
      "fieldName": "floatValue",
      "name": "Sum",
      "type": "doubleSum"
    }
  ],
  "filter": {
    "dimension": "type",
    "type": "selector",
    "value": "order.dessertSum"
  },
  "granularity": "day",
  "queryType": "timeseries"
}
```

**Reply by Pernic** [reply]
thanks for the response, I will do it like that.
